masonry/src/passes/update.rs

```diff
@@ -145,3 +146,52 @@ pub(crate) fn run_update_pointer_pass(root: &mut RenderRoot, root_state: &mut Wi
     // Pass root widget state to synthetic state create at beginning of pass
     root_state.merge_up(root.widget_arena.get_state_mut(root.root.id()).item);
 }
+
+// ----------------
+
+fn update_anim_for_widget(
+    global_state: &mut RenderRootState,
+    mut widget: ArenaMut<'_, Box<dyn Widget>>,
+    mut state: ArenaMut<'_, WidgetState>,
+    elapsed_ns: u64,
+) {
+    let _span = widget.item.make_trace_span().entered();
+
+    if !state.item.needs_anim {
+        return;
+    }
+
+    if state.item.request_anim {
+        let mut ctx = LifeCycleCtx {
+            global_state,
+            widget_state: state.item,
+            widget_state_children: state.children.reborrow_mut(),
+            widget_children: widget.children.reborrow_mut(),
+        };
+        widget
+            .item
+            .lifecycle(&mut ctx, &LifeCycle::AnimFrame(elapsed_ns));
+    }
+
+    state.item.needs_anim = false;
+    state.item.request_anim = false;
```

Comment:
It's not clear to me why this code is here.
Currently, this stops any animation after one frame, if it requests a new frame in the handler.

What is the expected behaviour if an animation wants to last more than one frame?

Reply:
Yeah, I think I rushed this PR too much.